pkg/virt-launcher/virtwrap/converter/builder.go

```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package converter
+
+import (
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type configurator interface {
+	Configure(vmi *v1.VirtualMachineInstance, domain *api.Domain) error
+}
+
+type DomainBuilder struct {
+	configurators []configurator
+}
+
+func NewDomainBuilder() DomainBuilder {
+	return NewDomainBuilderWithConfigurators()
+}
+
+func NewDomainBuilderWithConfigurators(configurators ...configurator) DomainBuilder {
+	return DomainBuilder{configurators: configurators}
+}
+
+func (db *DomainBuilder) BuildDomainFromVMI(vmi *v1.VirtualMachineInstance, domain *api.Domain) error {
```

Comment:
Two points:
1. Can we pass the VMI and the initial domain as part of the state? Does it have any usefulness as part of the method input?
2. The method name has redundant information IMO. `domain` is part of the bind struct name and the `vmi` is part of the signature. If we consider prev point, it will be trivial these extra naming is not needed.